axelor-base/src/main/resources/views/EventsPlanning.xml

```diff
@@ -10,11 +10,38 @@
 
   <form name="events-planning-form" title="Public holidays planning"
     model="com.axelor.apps.base.db.EventsPlanning">
+    <toolbar>
+      <button title="Add a period" name="addPeriodBtn"
+        onClick="save,action-events-planning-view-add-period-wizard"/>
+    </toolbar>
     <panel name="mainPanel">
       <field name="name"/>
     </panel>
     <panel-related name="eventsPlanningLineListPanel" field="eventsPlanningLineList"
       editable="true" grid-view="events-planning-line-grid" form-view="events-planning-line-form"/>
   </form>
 
+  <form name="add-period-wizard-form" model="com.axelor.utils.db.Wizard" title="Add a period"
+    onSave="action-method-generate-event-planning-lines">
+    <panel name="mainPanel">
+      <field name="$startDate" type="date" title="Start date" required="true"/>
```

Comment:
Hello,
Please do not use the old dirty check behavior `$` for dummy field, it is deprecated and will be removed in next major release of AOP. You may instead use `x-dirty`. (See https://redmine.axelor.com/news/3022)